samcli/lib/providers/sam_function_provider.py

```diff
@@ -446,7 +446,9 @@ def _build_function_configuration(
             codeuri = SamLocalStackProvider.normalize_resource_path(stack.location, codeuri)
 
         package_type = resource_properties.get("PackageType", ZIP)
-        if package_type == ZIP and resource_properties.get("Handler") is None:
+        if package_type == ZIP and (
+            resource_properties.get("Handler") is None or resource_properties.get("Handler") == ""
+        ):
```

Comment:
nit: this could be changed to a false-y check (not sure if we also wanted to validate type)